components/inputs/input-color.js

```diff
@@ -0,0 +1,411 @@
+					background-color: var(--d2l-color-mica);
+				}
+				:host([disabled]) button {
+					cursor: default;
+					opacity: 0.5;
+				}
+				/* Firefox includes a hidden border which messes up button dimensions */
+				button::-moz-focus-inner {
+					border: 0;
+				}
+
+				.swatch {
+					border: 1px inset rgba(0, 0, 0, 0.42);
+					border-radius: 0.1rem;
+					box-sizing: border-box;
+					display: inline-block;
+					height: 0.3rem;
+					width: 1.2rem;
+				}
+				:host([type="custom"]) .swatch {
+					border-radius: 0.15rem;
+					height: 1rem;
+				}
+				.swatch-transparent {
+					background-image: url("data:image/svg+xml;base64,${unsafeCSS(btoa(SWATCH_TRANSPARENT))}");
+					background-position-y: -1.5px;
+					background-size: cover;
+				}
+				:host([type="custom"]) .swatch-transparent {
+					background-position-y: 0;
+				}
+
+				.icon-wrapper {
+					align-items: center;
+					display: flex;
+					flex-direction: column;
+					gap: 0.1rem;
+				}
+				.icon-wrapper > svg {
+					height: 0.6rem;
+				}
+
+				.readonly-wrapper {
+					border-radius: 0.3rem;
+					display: inline-block;
+					line-height: 0;
+					outline: none;
+					padding: 0.55rem;
+				}
+				.readonly-wrapper:focus {
+					outline: none;
+				}
+				.readonly-wrapper:${unsafeCSS(getFocusPseudoClass())} {
+					box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine);
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.disallowNone = false;
+		this.labelHidden = false;
+		this.launchType = 'dialog';
+		this.readonly = false;
+		this.type = 'custom';
+		this._associatedValue = undefined;
+		this._missingLabelErrorHasBeenThrown = false;
+		this._opened = false;
+		this._validatingLabelTimeout = null;
+		this._value = undefined;
+	}
+
+	get associatedValue() { return this._associatedValue; }
+	set associatedValue(val) {
+		if (val !== undefined && !HEX_REGEX.test(val)) {
+			throw new TypeError(`<d2l-input-color>: invalid HEX associated-value "${val}"`);
+		}
+		const oldVal = this._associatedValue;
+		this._associatedValue = (typeof val === 'string') ? val.toUpperCase() : undefined;
+		this.requestUpdate('associatedValue', oldVal);
+	}
+
+	get value() {
+		if (this._value === undefined && this.disallowNone) {
+			return this.type === 'background' ? DEFAULT_VALUE_BG : DEFAULT_VALUE;
+		}
+		return this._value;
+	}
+	set value(val) {
+		if (val !== undefined && !HEX_REGEX.test(val)) {
+			throw new TypeError(`<d2l-input-color>: invalid HEX value "${val}"`);
+		}
+		const oldVal = this._value;
+		this._value = (typeof val === 'string') ? val.toUpperCase() : undefined;
+		this.requestUpdate('value', oldVal);
+	}
+
+	static get focusElementSelector() {
+		return '#opener';
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._validateLabel();
+	}
+
+	render() {
+
+		const label = !this.labelHidden ? html`<div class="d2l-input-label">${this._getLabel()}</div>` : nothing;
```

Comment:
Unlike with other inputs this "label" is just "for show", since technically this is a button and not an input and the button uses a combination of the tooltip and `aria-label` as its label.